### PR TITLE
Buildsystem changes

### PR DESCRIPTION
**Pull request by a17r**
Today I was working on Gentoo's oyranos live ebuild for git master and along the way created a new buildsystem patch.

- When distros specifically remove bundled libs as part of their policy, configure should not fall back to using bundled libs and rather bail out when a requested dependency can't be found
- Use cmake-provided FindCups.cmake (no need to ship FindCUPS.cmake then, imo)*

\* same could be done for FindFltk and FindXcm, Gentoo had been removing those files already but I didn't look if there was a specific reason for them being shipped with oyranos

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -48,8 +48,8 @@ ENDIF()
 FIND_PROGRAM(CMAKE_AR_TMP NAMES ar)
 
 # detect top level projects
-FIND_PACKAGE( Yajl )
```
Comment (beku) [inline CMakeLists.txt]: Would the following meet your needs?:
IF(USE_SYSTEM_YAJL)
  FIND_PACKAGE(Yajl REQUIRED)
ELSE()
  FIND_PACKAGE( Yajl )
ENDIF()
IF(YAJL_FOUND)
  ... as is ...

The reason is, the config stage shall not fail if user give no -DUSE_SYSTEM_YAJL. I consider it optional.

Comment (a17r) [inline CMakeLists.txt]: Yeah, that would do the trick as well. ;)

My patch would have led to bundled yajl and elektra being used when someone hadn't set -DUSE_SYSTEM_ instead.

**Comment by a17r** [comment]
...another thing the existing Gentoo patch does is appending "-${OYRANOS_VERSION}" to the doc install location. Most other packages I've come across seem to do that per default, but it's a minor thing for me.

**Comment by beku** [comment]
would good upstream too, I guess. However, let us merge this now.

**Pull request merged**